hydra-node/src/Hydra/Chain/CardanoClient.hs

```diff
@@ -167,15 +187,16 @@ awaitTransaction ::
   NetworkId ->
   -- | Filepath to the cardano-node's domain socket
   SocketPath ->
+  -- | The current running era we can use to query the node
+  CardanoEra era ->
   -- | The transaction to watch / await
   Tx ->
   IO UTxO
-awaitTransaction networkId socket tx =
-  go
+awaitTransaction networkId socket era tx = go
  where
   ins = keys (UTxO.toMap $ utxoFromTx tx)
   go = do
-    utxo <- queryUTxOByTxIn networkId socket QueryTip ins
+    utxo <- queryUTxOByTxIn networkId socket QueryTip ins era
```

Comment:
Should: Match `queryUTxOByTxIn` the signature of `queryUTxO` (not require an `era)

This is inconsistent and by removing the era (query the current era in the same node connection), we can avoid quite some changes on the diff.

Reply:
nice catch!